### PR TITLE
Multi client subscribeNewHead

### PR DESCRIPTION
**Pull request by markya0616**

### CODE REVIEW

File: multiclient/option.go [hunk 1]
```diff
@@ -29,18 +29,22 @@ type Option func(*Client) error
...
-		mc.ethURLs = append(mc.ethURLs, ethURLs...)
+		for _, url := range ethURLs {
+			mc.rpcClientMap.Set(url, nil)
+		}
 		return nil
 	}
 }
 
-func ConsulDiscocvery(rawURL, serviceID, serviceScheme string) Option {
+func ConsulDiscovery(rawURL, serviceID, serviceScheme string) Option {
```
Comment (tailingchen) [inline multiclient/option.go:39]: Nice!

File: multiclient/client.go [hunk 2]
```diff
@@ -37,70 +38,72 @@ const (
...
+	errCh := make(chan error, lens)
+	clients := mc.rpcClientMap.Map()
+	for rawURL := range clients {
+		go func(ctx context.Context, rawURL string) {
 			ctx, cancel := context.WithTimeout(ctx, dialTimeout)
 			defer cancel()
 			c, err := rpc.DialContext(ctx, rawURL)
 			if err != nil {
 				log.Error("Failed to dial eth client", "rawURL", rawURL, "err", err)
 			}
```
Comment (tailingchen) [inline multiclient/client.go:77]: Would like to add one info log.
```
log.Info("Connect to ethclient successfully", "url", rawURL)
```

File: multiclient/client.go [hunk 3]
```diff
@@ -520,6 +538,56 @@ func (mc *Client) BatchCallContext(ctx context.Context, isPostToAll bool, b []rp
...
 }
 
+// Subscribe API
+
+// SubscribeNewHead subscribes to notifications about the current blockchain head
+// on the given channel.
+func (mc *Client) SubscribeNewHead(ctx context.Context, ch chan<- *types.Header) (ethereum.Subscription, error) {
+	clientsMap := mc.rpcClientMap.Map()
+	lens := len(clientsMap)
+	if lens == 0 {
```
Comment (tailingchen) [inline multiclient/client.go:594]: It's a good check before perform rpc call.
It's better to add check for each method.

Comment (markya0616) [reply multiclient/client.go:594]: OK

File: multiclient/client.go (hunk 2 above)
Comment (markya0616) [reply multiclient/client.go:77]: OK

**Review by tailingchen — APPROVED** [review]

**Pull request merged**